### PR TITLE
Add Index tags + small updates to generated code

### PR DESCRIPTION
**Pull request by aulorbe**
## Problem

This is a late addition to the current version of our client to incorporate Index Tags, as introduced in the `2024-10` spec. 

Once merged, this will be a `minor` release of the current version.

Note: This PR also contains some generated code that was added into the `2024-10` spec after the latest code gen on `main`.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

CI passes w/new tests and additions to old tests.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208820098587320

### CODE REVIEW

File: src/control/__tests__/configureIndex.test.ts [hunk 1]
```diff
@@ -25,21 +25,35 @@ describe('configureIndex', () => {
         ready: true,
         state: 'Ready',
       },
+      tags: {
+        example: 'tag',
+      },
+      deletionProtection: 'disabled', // Redundant, but for example purposes
```
Comment (aulorbe) [inline src/control/__tests__/configureIndex.test.ts:31]: Added `deletionProtection` to this test just b/c I noticed it was missing while doing index tag stuffs

File: src/control/createIndex.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ const CreateIndexOptionsProperties: CreateIndexOptionsType[] = [
   'deletionProtection',
   'waitUntilReady',
   'suppressConflicts',
+  'tags',
```
Comment (austin-denoble) [inline src/control/createIndex.ts:38]: Would it be worth adding any `createIndex` unit test coverage for this? I know you're integration testing as well, so maybe not.

**Review by austin-denoble — APPROVED** [review]
LGTM, thanks for getting this worked in!

File: README.md [hunk 3]
```diff
@@ -344,6 +346,31 @@ await pc.configureIndex('deletion-protected-index', {
 });
 ```
 
+#### Create or configure an index with index tags
+
+You can configure both serverless and pod indexes with [tags](https://docs.pinecone.io/guides/indexes/tag-an-index).
+Indexes can hold an arbitrary number of tags outlining metadata
+you would like attached to the index object itself, such as team ownership, project, or any other relevant information.
```
[the remark this answers is not in the document]
Comment (aulorbe) [reply README.md:353]: Sure, I can add that!

File: src/control/createIndex.ts (hunk 2 above)
Comment (aulorbe) [reply src/control/createIndex.ts:38]: Wouldn't hurt

File: README.md [hunk 4]
```diff
@@ -344,6 +346,31 @@ await pc.configureIndex('deletion-protected-index', {
 });
 ```
 
+#### Create or configure an index with index tags
+
+You can configure both serverless and pod indexes with [tags](https://docs.pinecone.io/guides/indexes/tag-an-index).
```
[the remark this answers is not in the document]
Comment (aulorbe) [reply README.md:351]: ✅

**Pull request merged**